Compilation Error on Serialization Sample

The serialization sample doesn't compile on my computer. Here's the error:

```
core/include/cubos/memory/stream.hpp:229:28: error: call of overloaded 'print(long unsigned int&)' is ambiguous
  229 |                 this->print(arg);
      |                 ~~~~~~~~~~~^~~~~
cubos/core/include/cubos/memory/stream.hpp:71:14: note: candidate: 'void cubos::memory::Stream::print(int8_t, size_t)'
   71 |         void print(int8_t value, size_t base = 10);
      |              ^~~~~
cubos/core/include/cubos/memory/stream.hpp:76:14: note: candidate: 'void cubos::memory::Stream::print(int16_t, size_t)'
   76 |         void print(int16_t value, size_t base = 10);
      |              ^~~~~
cubos/core/include/cubos/memory/stream.hpp:81:14: note: candidate: 'void cubos::memory::Stream::print(int32_t, size_t)'
   81 |         void print(int32_t value, size_t base = 10);
      |              ^~~~~
cubos/core/include/cubos/memory/stream.hpp:86:14: note: candidate: 'void cubos::memory::Stream::print(int64_t, size_t)'
   86 |         void print(int64_t value, size_t base = 10);
      |              ^~~~~
cubos/core/include/cubos/memory/stream.hpp:91:14: note: candidate: 'void cubos::memory::Stream::print(uint8_t, size_t)'
   91 |         void print(uint8_t value, size_t base = 10);
      |              ^~~~~
cubos/core/include/cubos/memory/stream.hpp:96:14: note: candidate: 'void cubos::memory::Stream::print(uint16_t, size_t)'
   96 |         void print(uint16_t value, size_t base = 10);
      |              ^~~~~
cubos/core/include/cubos/memory/stream.hpp:101:14: note: candidate: 'void cubos::memory::Stream::print(uint32_t, size_t)'
  101 |         void print(uint32_t value, size_t base = 10);
      |              ^~~~~
cubos/core/include/cubos/memory/stream.hpp:106:14: note: candidate: 'void cubos::memory::Stream::print(uint64_t, size_t)'
  106 |         void print(uint64_t value, size_t base = 10);
      |              ^~~~~
cubos/core/include/cubos/memory/stream.hpp:111:14: note: candidate: 'void cubos::memory::Stream::print(float, size_t)'
  111 |         void print(float value, size_t decimalPlaces = 4);
      |              ^~~~~
cubos/core/include/cubos/memory/stream.hpp:116:14: note: candidate: 'void cubos::memory::Stream::print(double, size_t)'
  116 |         void print(double value, size_t decimalPlaces = 4);
      |              ^~~~~
```

--- a/core/include/cubos/memory/stream.hpp
+++ b/core/include/cubos/memory/stream.hpp
@@ -65,40 +65,24 @@ namespace cubos::memory
         /// @param c The byte to put.
         void put(char c);
 
-        /// Prints a 8 bit signed integer to the stream.
+        /// Prints a signed integer to the stream.
+        /// @tparam T The type of the integer.
         /// @param value The value to print.
         /// @param base The base to use.
-        void print(int8_t value, size_t base = 10);
+        template <typename T>
+        requires(std::is_integral_v<T>&& std::is_signed_v<T>) void print(T value, size_t base = 10);
 
-        /// Prints a 16 bit signed integer to the stream.
+        /// Prints a signed integer to the stream.
+        /// @tparam T The type of the integer.
         /// @param value The value to print.
         /// @param base The base to use.
-        void print(int16_t value, size_t base = 10);
-
-        /// Prints a 32 bit signed integer to the stream.
-        /// @param value The value to print.
-        /// @param base The base to use.
-        void print(int32_t value, size_t base = 10);
+        template <typename T>
+        requires(std::is_integral_v<T> && !std::is_signed_v<T>) inline void print(T value, size_t base = 10);
 
         /// Prints a 64 bit signed integer to the stream.
         /// @param value The value to print.
         /// @param base The base to use.
         void print(int64_t value, size_t base = 10);
-
-        /// Prints a 8 bit unsigned integer to the stream.
-        /// @param value The value to print.
-        /// @param base The base to use.
-        void print(uint8_t value, size_t base = 10);
-
-        /// Prints a 16 bit unsigned integer to the stream.
-        /// @param value The value to print.
-        /// @param base The base to use.
-        void print(uint16_t value, size_t base = 10);
-
-        /// Prints a 32 bit unsigned integer to the stream.
-        /// @param value The value to print.
-        /// @param base The base to use.
-        void print(uint32_t value, size_t base = 10);
 
         /// Prints a 64 bit unsigned integer to the stream.
         /// @param value The value to print.
@@ -212,6 +196,18 @@ namespace cubos::memory
     };
 
     // Implementation
+
+    template <typename T>
+    requires(std::is_integral_v<T>&& std::is_signed_v<T>) inline void Stream::print(T value, size_t base)
+    {
+        this->print(static_cast<int64_t>(value), base);
+    }
+
+    template <typename T>
+    requires(std::is_integral_v<T> && !std::is_signed_v<T>) inline void Stream::print(T value, size_t base)
+    {
+        this->print(static_cast<uint64_t>(value), base);
+    }
 
     template <typename T, typename... TArgs> void Stream::printf(const char* fmt, T arg, TArgs... args)
     {

--- a/core/src/cubos/memory/stream.cpp
+++ b/core/src/cubos/memory/stream.cpp
@@ -31,21 +31,6 @@ void Stream::put(char c)
     this->write(&c, sizeof(c));
 }
 
-void Stream::print(int8_t value, size_t base)
-{
-    this->print(static_cast<int64_t>(value), base);
-}
-
-void Stream::print(int16_t value, size_t base)
-{
-    this->print(static_cast<int64_t>(value), base);
-}
-
-void Stream::print(int32_t value, size_t base)
-{
-    this->print(static_cast<int64_t>(value), base);
-}
-
 void Stream::print(int64_t value, size_t base)
 {
     if (value < 0)
@@ -57,20 +42,6 @@ void Stream::print(int64_t value, size_t base)
     this->print(static_cast<uint64_t>(value), base);
 }
 
-void Stream::print(uint8_t value, size_t base)
-{
-    this->print(static_cast<uint64_t>(value), base);
-}
-
-void Stream::print(uint16_t value, size_t base)
-{
-    this->print(static_cast<uint64_t>(value), base);
-}
-
-void Stream::print(uint32_t value, size_t base)
-{
-    this->print(static_cast<uint64_t>(value), base);
-}
 
 void Stream::print(uint64_t value, size_t base)
 {

